### PR TITLE
added fields in GHPullRequestReviewComment which were missing

### LINKED ISSUE #1463: No start_line/line in GHPullRequestReviewComment
**Describe the bug**
https://docs.github.com/en/rest/pulls/comments#get-a-review-comment-for-a-pull-request
Github states that review comment has following fields:
```
 "start_line": {
        "type": [
          "integer",
          "null"
        ],
        "description": "The first line of the range for a multi-line comment.",
        "examples": [
          2
        ]
      },
      "original_start_line": {
        "type": [
          "integer",
          "null"
        ],
        "description": "The first line of the range for a multi-line comment.",
        "examples": [
          2
        ]
      },
      "start_side": {
        "type": [
          "string",
          "null"
        ],
        "description": "The side of the first line of the range for a multi-line comment.",
        "enum": [
          "LEFT",
          "RIGHT",
          null
        ],
        "default": "RIGHT"
      },
      "line": {
        "description": "The line of the blob to which the comment applies. The last line of the range for a multi-line comment",
        "type": "integer",
        "examples": [
          2
        ]
      },
      "original_line": {
        "description": "The line of the blob to which the comment applies. The last line of the range for a multi-line comment",
        "type": "integer",
        "examples": [
          2
        ]
      },
```
But there are no such fields in current API (class `GHPullRequestReviewComment`)

**Expected behavior**
GHPullRequestReviewComment contains all fields which are provided by GitHub

**Additional context**
Thank you for wonderful library! ❤️

### PR DESCRIPTION
**Pull request by kisaga**
# Description

added new fields in GHPullRequestReviewComment that were missing. 
Assuming that the changes are minor the mocked data snapshot has not been updated.

The response and its JSON Schema can be found [here](https://docs.github.com/en/rest/pulls/comments#get-a-review-comment-for-a-pull-request)
fixes #1463

# Before submitting a PR:

- [x] Changes must not break binary backwards compatibility. If you are unclear on how to make the change you think is needed while maintaining backward compatibility, [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [x] Add JavaDocs and other comments as appropriate. Consider including links in comments to relevant documentation on https://docs.github.com/en/rest .  
- [x] Add tests that cover any added or changed code. This generally requires capturing snapshot test data. See [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.
- [x] Push your changes to a branch other than `main`. You will create your PR from that branch.

# When creating a PR: 

- [x] Fill in the "Description" above with clear summary of the changes. This includes:
- [x] If this PR fixes one or more issues, include "Fixes #<issue number>" lines for each issue. 
- [x] Provide links to relevant documentation on https://docs.github.com/en/rest where possible.
- [x] All lines of new code should be covered by tests as reported by code coverage. Any lines that are not covered must have PR comments explaining why they cannot be covered. For example, "Reaching this particular exception is hard and is not a particular common scenario."
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by gsmet** [comment]
> Assuming that the changes are minor the mocked data snapshot has not been updated.

I think this is gonna be a problem. We need to be able to regenerate the mocked data and still have the tests passing. You should probably create a test project in the test org, and add a review there that allows to test this thing.

**Comment by gsmet** [comment]
> Assuming that the changes are minor the mocked data snapshot has not been updated.

I think this is gonna be a problem. We need to be able to regenerate the mocked data and still have the tests passing. You should probably create a test project in the test org, and add a review there that allows to test this thing.

**Comment by kisaga** [comment]
Cool! I'll check how to create a project in the test org and how to update the mock data.

**Comment by kisaga** [comment]
Hi @gsmet I've run the tests against the https://github.com/hub4j-test-org/github-api twice:

- https://github.com/hub4j-test-org/github-api/pull/451
- https://github.com/hub4j-test-org/github-api/pull/452

And I have some questions:

1. In the Contributing guide it is mentioned `Please request access to this org to record your tests before submitting a PR.`. From whom should I ask access ? The two tests I've run are failing at the validation of AuthorAssociation field because mine is NONE but the expected is MEMBER.
2. Is it a problem if I continue using this project to modify the setup phase of the test I've modified  ? (I need to add some reactions to get the respective field in the response) 
3. For the final snapshot that will be committed what should be the target repository for the responses ? 

Waiting for your feedback :slightly_smiling_face:

**Comment by kisaga** [comment]
Hi @gsmet,
I managed to take a snapshot after modifying the test to add some reactions, fetch again the comment and check the reactions. I've also modified some values that I had hardcoded in the mock responses in my initial approach.
The snapshot is from [this PR](https://github.com/hub4j-test-org/github-api/pull/455) and I have modified manually the author_association from NONE to MEMBER.
I don't know if the snapshot is fine or if it should be taken as a MEMBER, but even if this the case, if I become member of the test org I can take a new snapshot and commit the changes.

Let me know if it is fine or if I should take the snapshot as a MEMBER (if yes, how can I join the test org).

**Comment by bitwiseman** [comment]
@kisaga 
I've invited you to the test org.

File: src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java [hunk 1]
```diff
@@ -43,16 +44,27 @@
 public class GHPullRequestReviewComment extends GHObject implements Reactable {
     GHPullRequest owner;
 
+    private Long pull_request_review_id = -1L;
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java:47]:
```suggestion
    private long pull_request_review_id = -1L;
```

File: src/main/java/org/kohsuke/github/GHPullRequestReviewCommentReactions.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+/**
+ * Reactions for a Pull Request Review comment.
+ *
+ * @author Vasilis Gakias
+ * @see <a href="https://docs.github.com/en/rest/pulls/comments#get-a-review-comment-for-a-pull-request">API
+ *      documentation</a>
+ * @see GHPullRequestReviewComment
+ */
+public class GHPullRequestReviewCommentReactions {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPullRequestReviewCommentReactions.java:15]: I am not seeing the `reactions` field listed in the doc link mentioned here.  
This page talks about reactions API: 
https://docs.github.com/en/rest/reactions#list-reactions-for-a-pull-request-review-comment

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Please double check that the `reactions` field is actually part of the supported API and not deprecated and soon to be removed.

**Review by kisaga — COMMENTED** [review]
#### Avoiding `reactions` field deprecation
I've checked [here](https://github.blog/changelog/label/deprecation/) and I couldn't see any deprecation notice regarding the `reactions` field. 
I've also [queried](https://twitter.com/search?q=reactions%20(from%3AGHchangelog)&src=typed_query&f=top) twitter for any reaction related changes but I didn't find anything related except for [this](https://github.blog/changelog/2022-02-11-legacy-delete-reactions-rest-api-removed/) which is not related to these changes.

#### Data snapshot as a member
I've also updated the snapshot with a membership and the files are fine now.

#### Pending
Let me know if we want to change the `pull_request_review_id` from `Long` to `long` with the side effect of having 0 default value instead of -1 if the value the response is null. I've found [this](https://stackoverflow.com/questions/14434679/how-do-i-make-jackson-not-serialize-primitives-with-default-value) but I couldn't apply it and also could not find similar in this project

File: src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java (hunk 1 above)
Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java:47]: :shrug: It's fine.  Thanks for explaining.

**Comment by bitwiseman** [comment]
💢 So, the `reactions` field shows up in the "response schema" on that section but not in the "example response". :(

```
...
    "reactions": {
      "title": "Reaction Rollup",
      "type": "object",
      "properties": {
        "url": {
          "type": "string",
          "format": "uri"
        },
        "total_count": {
          "type": "integer"
        },
        "+1": {
          "type": "integer"
        },
        "-1": {
          "type": "integer"
        },
        "laugh": {
          "type": "integer"
        },
        "confused": {
          "type": "integer"
        },
        "heart": {
          "type": "integer"
        },
        "hooray": {
          "type": "integer"
        },
        "eyes": {
          "type": "integer"
        },
        "rocket": {
          "type": "integer"
        }
      },
      "required": [
        "url",
        "total_count",
        "+1",
        "-1",
        "laugh",
        "confused",
        "heart",
        "hooray",
        "eyes",
        "rocket"
      ]
    },
...
```

**Comment by kisaga** [comment]
HI @bitwiseman I've asked a question [here](https://github.com/github-community/community/discussions/19152).
We can wait for an answer and then make our decision for this field.
If we need the rest of the changes we can:
- either include the reactions field and remove it if the answer will be that it'll removed at some point. In this case if it will be deprecated it will have a null value
- or exclude it and add it in case the answer is that it will not be removed

**Comment by bitwiseman** [comment]
@kisaga Please make the change requested above so we can merge the rest of this PR.

File: src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java [hunk 3]
```diff
@@ -227,6 +239,115 @@ protected String getApiRoute(boolean includePullNumber) {
...
+    /**
+     * Gets the Reaction Rollup
+     *
+     * @return {@link GHPullRequestReviewCommentReactions} the reaction rollup
+     */
+    public GHPullRequestReviewCommentReactions getReactions() {
+        return reactions;
+    }
+
+    public static enum Side {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java:342]:
```suggestion
   /**
    * The side of the diff to which the comment applies
    */
    public static enum Side {
```

File: src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java [hunk 4]
```diff
@@ -227,6 +239,118 @@ protected String getApiRoute(boolean includePullNumber) {
...
+     *
+     * @return {@link GHPullRequestReviewCommentReactions} the reaction rollup
+     */
+    public GHPullRequestReviewCommentReactions getReactions() {
+        return reactions;
+    }
+
+   /**
+    * The side of the diff to which the comment applies
+    */
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java:344]:
```suggestion
    /**
     * The side of the diff to which the comment applies
     */
```

File: src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java [hunk 5]
```diff
@@ -227,6 +239,118 @@ protected String getApiRoute(boolean includePullNumber) {
...
+
+    /**
+     * The side of the diff to which the comment applies
+     */
+    public static enum Side {
+        RIGHT, LEFT, UNKNOWN;
+
+        public static Side from(String value) {
+            return EnumUtils.getEnumOrDefault(Side.class, value, Side.UNKNOWN);
+        }
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java:350]:
```suggestion
    /**
     * The side of the diff to which the comment applies
     */
    public static enum Side {
        
        /** Right side */
        RIGHT,
        /** Left side */
        LEFT,
        /** Unknown side */
        UNKNOWN;

        /**
         * From.
         *
         * @param value
         *            the value
         * @return the status
         */
        public static Side from(String value) {
            return EnumUtils.getEnumOrDefault(Side.class, value, Side.UNKNOWN);
        }
```

File: src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java [hunk 6]
```diff
@@ -227,6 +239,131 @@ protected String getApiRoute(boolean includePullNumber) {
...
+     */
+    public GHPullRequestReviewCommentReactions getReactions() {
+        return reactions;
+    }
+
+    /**
+     * The side of the diff to which the comment applies
+     */
+    public static enum Side {
+
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPullRequestReviewComment.java:346]:
```suggestion
```

**Pull request merged**